tapfeatures/aux_channel_negotiator.go

```diff
@@ -0,0 +1,191 @@
+package tapfeatures
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnutils"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+// AuxChannelNegotiator is responsible for producing the extra tlv blob that is
+// encapsulated in the init and reestablish peer messages. This helps us
+// communicate custom feature bits with our peer.
+type AuxChannelNegotiator struct {
+	// peerFeatures keeps track of the supported features of our peers. This
+	// map will be used for lookups by other subsystems, when some features
+	// need to be supported by both parties to take effect.
+	peerFeatures lnutils.SyncMap[route.Vertex, *lnwire.RawFeatureVector]
```

Comment:
Ah ok, I hadn't reviewed this PR yet before reviewing the follow up. I see now that the negotiator stores the information retrieved from `lnd` directly in memory.